core/src/main/java/com/linecorp/armeria/common/thrift/ThriftCallbackCompletableFuture.java

```diff
@@ -0,0 +1,22 @@
+package com.linecorp.armeria.common.thrift;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.thrift.async.AsyncMethodCallback;
+
+/**
+ * A {@link CompletableFuture} that can be passed in as an {@link AsyncMethodCallback}
+ * when making an asynchronous thrift client rpc.
+ */
+public class ThriftCallbackCompletableFuture<T> extends CompletableFuture<T> implements AsyncMethodCallback<T> {
+
```

Comment:
How about adding two more constructors (successful and failed) as well as the default one for easier construction?